model/task_lifecycle.go

```diff
@@ -830,7 +830,7 @@ func HandleEndTaskForCommitQueueTask(t *task.Task, status string) error {
 		return errors.Errorf("no commit queue found for '%s'", t.Project)
 	}
 	if status != evergreen.TaskSucceeded && !t.Aborted {
-		return dequeueAndRestartWithStepback(cq, t, evergreen.APIServerTaskActivator, fmt.Sprintf("task '%s' failed", t.DisplayName))
+		return dequeueAndRestartWithStepback(cq, t, evergreen.MergeTestRequester, fmt.Sprintf("task '%s' failed", t.DisplayName))
```

Comment:
It looks like fixing this will make sure that scheduled task get unscheduled. Will it also abort in progress tasks?

Reply:
Yes, it will call [DequeueAndRestartForTask](DequeueAndRestartForTask), which later calls [CancelPatch](https://github.com/evergreen-ci/evergreen/blob/main/model/patch_lifecycle.go#L778) which has a line to abort the in-progress tasks